### DISCUSSION TITLE
Program in Venv works as expected, when compiled, gives a 'flashing console' bug.

### DISCUSSION THREAD

**Discussion by nednoodlehead** [pyinstaller/pyinstaller#7054] [Help]
I have a music app that uses FFMPEG. Not a library, but a path to the .exe, and a Tkinter GUI. 

When compiled, each time I use a function that calls an FFMPEG operation, window's cmd will quickly open then close. This does not happen while using it inside of the virtual environment.

I have a github repo that can be used to recreate this bug (with steps included) here: https://github.com/nednoodlehead/ffmpegthreading

I also have a quick video showing what this bug looks like:
https://www.youtube.com/watch?v=tCZ0z9E5Iyw

A few things to note with this:

1). This will only occur when the app is compiled. Running this in a virtual environment works as expected. I run it from my IDE all the time with no console flashing. (as seen in video)

2). It will only flash while a GUI is present. So using a regular script to call the same command that produces the bug will NOT cause the bug

3). The bug was tested in Tkinter and PyQt5 and produced the same result (flashing console)

4). Threading is NOT needed to reproduce this bug, it just makes the debugging a bit easier

5). Using the ffmpeg module does not work as my entire project is based around an interface for PyDub (Which uses ffmpeg)

6). This solution does not work for me: [How to hide console output of FFmpeg in Python?](https://stackoverflow.com/questions/62968888/how-to-hide-console-output-of-ffmpeg-in-python/71741286#71741286) (Because this uses the module, while I use the exe)

7). I am using auto-py-to-exe to compile (which uses pyinstaller)

8). Running 'ffmpeg -version' (in regular cmd (same result in venv)) gives: ffmpeg version N-106605-gf67403edb3-20220413

9). This has NOT been tested on Linux or Mac, I would appreciate if someone could test it because after asking around a bit, it seems likely that this is a windows bug in how windows handles the processes being called

10). My code uses a library called 'Pydub' which calls py-Simple-Audio, which calls upon the FFMPEG exe. (Convoluted, i know)

**Comment by rokm** [comment]
If you try to run a console-enabled process from an application that has no console available, console window is opened.

The only reason the "program works in venv as expected" is because you're running it through regular `python.exe` interpreter, which provides the console and so your ffmpeg subprocess does not need to open one. Try running running it using the noconsole interpreter, `pythonw.exe`, and you'll see that console window pop up momentarily (presumably for duration of audio file transcoding?) as well.

**Reply by nednoodlehead** [reply]
Yeah I see what you mean. Running pythonw.exe makes it bug, while python.exe does not. I'm not well versed in how pyinstaller exactly works, but is there a way I can use the python.exe way instead of pythonw.exe ? Is that possible?

**Reply by rokm** [reply]
PyInstaller builds in console mode by default; since you're using auto-py-to-exe, you probably need to check the setting under "Console Window".

**Reply by nednoodlehead** [reply]
Yup. That's it. Choosing 'Console Based' under 'Console Window' tab fixes that issue. Only problem is that a cmd prompt is just chilling there. Shouldn't be too hard to send it to tray though. 
Thanks a bunch, knew it would be something simple
